### ISSUE TITLE
Use the new vscode.workspace.fs API

### ISSUE THREAD

**Issue by robertohuertasm** [vscode-icons/vscode-icons#2199]
See [vscode docs](https://code.visualstudio.com/updates/v1_37#_vscodeworkspacefs). As per Microsoft's recommendation: 

> Call to Action: If your extension is currently using the fs module from Node.js, consider migrating to the new vscode.workspace.fs API.

**Comment by JimiC** [comment]
Can't seem to find the reasoning. Any links?

**Comment by robertohuertasm** [comment]
Check the link in the description.

**Comment by JimiC** [comment]
I did. The docs doesn't say the why.

**Comment by robertohuertasm** [comment]
it allows access to files from contributed file systems, such as the SSH or WSL remote file systems.

**Comment by JimiC** [comment]
That's no serious reason to migrate.

**Comment by robertohuertasm** [comment]
I'm not saying this is urgent either. I just left this here for our consideration given that Microsoft suggested to use this new API. Of course, this is not a matter of life or death, but it may be worth it to explore the new API. If it's just a wrapper around `fs` (I don't have a clue at the moment) then it shouldn't be difficult to integrate it.

**Comment by JimiC** [comment]
Easier said than done.

**Comment by KingDarBoja** [comment]
Today I was looking at this API and the available method equivalent to the ones already available at `fsAsync` file.

There is no `rmdir` or `unlink` method but I do feel like the `delete` one fullfill its purpose.

![image](https://user-images.githubusercontent.com/30560560/62826350-91a10380-bb7f-11e9-97f4-3baea00bb7e6.png)

Still haven't tested it, I am playing around at the source code.

**Comment by JimiC** [comment]
`vscode API` is already asynchronous so don't need `promisify`.

**Comment by JimiC** [comment]
And since you are going to use `vscode API` those functions need to be moved to `VSCodeManager` class and all tests have to be changed.

As I said... Easier said than done.

**Comment by KingDarBoja** [comment]
@JimiC Yeah, you are right, without mention that you have to pass for each method a `Uri` object rather that plain string.

Due to that, that's not going to be easy to migrate so I do think we should keep using the old `fs` from Node as JimiC stated.

**Comment by JimiC** [comment]
On another note, the switch to this API can only occur when our minimum `vscode` version reaches `v1.37`.

**Comment by xgdgsc** [comment]
https://github.com/microsoft/vscode/issues/82136 Please consider switching to new API as remote development become more popular.

**Comment by JimiC** [comment]
@xgdgsc Your linked issue has nothing to do with the use of the API or not.

**Comment by JimiC** [comment]
The minimum support `vscode` version has now reached `v1.31.1`. We are getting closer to `v1.37`.

**Comment by Alexstoppel** [comment]
Any updates on this?

**Comment by sasial-dev** [comment]
As explained [here](https://github.com/vscode-icons/vscode-icons/pull/2835#issuecomment-918255934), the maintainers have very little bandwith to be able to take on new projects!
